lib/fs.js

```diff
@@ -320,7 +321,7 @@ fs.existsSync = function(path) {
 };
 
 fs.readFile = function(path, options, callback) {
-  callback = maybeCallback(arguments[arguments.length - 1]);
+  callback = maybeCallback(callback || options);
```

Comment:
I think it's also because `arguments` was being used in a function where the named arguments were being overwritten. This could be a Crankshaft-specific limitation though, I'm not sure.